### PR TITLE
@TestPropertyOverride now throws FileNotFoundException for missing files

### PR DESCRIPTION
**Pull request by twicksell**

### CODE REVIEW

File: archaius2-test/src/test/java/com/netflix/archaius/test/Archaius2RuleFailureTest.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+import java.lang.annotation.Annotation;
+
+import org.junit.Test;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+
+public class Archaius2RuleFailureTest {
+
+    @Test(expected = TestConfigException.class)
```
Comment (qiangdavidliu) [inline archaius2-test/src/test/java/com/netflix/archaius/test/Archaius2RuleFailureTest.java]: expectation should be for FileNotFoundException

File: archaius2-test/src/main/java/com/netflix/archaius/test/TestPropertyOverrideAnnotationReader.java [hunk 2]
```diff
@@ -18,7 +20,14 @@ public Properties getPropertiesForAnnotation(TestPropertyOverride annotation) {
...
             try {
-                properties.load(this.getClass().getClassLoader().getResourceAsStream(fileName));
+                InputStream propFileStream = this.getClass().getClassLoader().getResourceAsStream(fileName);
+                if(propFileStream != null) {
+                    properties.load(propFileStream);
+                }
+                else {
+                    throw new FileNotFoundException(fileName);
+                }
+                properties.load(propFileStream);
```
Comment (twicksell) [inline archaius2-test/src/main/java/com/netflix/archaius/test/TestPropertyOverrideAnnotationReader.java]: Bad merge, thanks

File: archaius2-test/src/test/java/com/netflix/archaius/test/Archaius2RuleFailureTest.java (hunk 1 above)
Comment (twicksell) [inline archaius2-test/src/test/java/com/netflix/archaius/test/Archaius2RuleFailureTest.java]: It gets wrapped up in TestConfigException along the way

Comment (qiangdavidliu) [inline archaius2-test/src/test/java/com/netflix/archaius/test/Archaius2RuleFailureTest.java]: ah cool

**Pull request merged**